src/vm-vcpu-ref/src/x86_64/mptable.rs

```diff
@@ -148,17 +148,22 @@ pub struct MpTable {
 
 impl MpTable {
     /// Creates a new MP Table that can hold `cpu_num`.
-    pub fn new(cpu_num: u8) -> Result<MpTable> {
+    pub fn new(cpu_num: u8, max_irq: u8) -> Result<MpTable> {
         if cpu_num > MAX_SUPPORTED_CPUS {
             return Err(Error::TooManyCpus);
         }
 
         Ok(MpTable {
-            irq_num: IRQ_MAX + 1,
+            irq_num: max_irq + 1,
             cpu_num,
         })
     }
 
+    /// Returns the number of irqs
```

Comment:
I think this is no longer needed because we are returning the IRQ from the config. Same goes for the `irq_num` function in the `Gic`.